File: tests/inputs/hello.zig
// Sample Zig file to exercise syntax highlighting, indents,
// and folds in vorto. Open with `vorto assets/samples/hello.zig`.

const std = @import("std");

const Classification = enum {
    negative,
    zero,
    positive_even,
    positive_odd,

    fn label(self: Classification) []const u8 {
        return switch (self) {
            .negative => "negative",
            .zero => "zero",
            .positive_even => "positive even",
            .positive_odd => "positive odd",
        };
    }
};

const Person = struct {
    name: []const u8,
    age: u32,

    fn isAdult(self: Person) bool {
        return self.age >= 18;
    }
};

fn classify(n: i64) Classification {
    if (n < 0) return .negative;
    if (n == 0) return .zero;
    return if (@mod(n, 2) == 0) .positive_even else .positive_odd;
}

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const alice = Person{ .name = "Alice", .age = 30 };

    try stdout.print("Hello, {s}!\n", .{alice.name});

    var n: i64 = -2;
    while (n <= 5) : (n += 1) {
        try stdout.print("{d} -> {s}\n", .{ n, classify(n).label() });
    }
}

File: tests/inputs/hello.rs
// Sample Rust file to exercise syntax highlighting, indents,
// and folds in vorto. Open with `vorto assets/samples/hello.rs`.

use std::collections::HashMap;
use std::fmt;

/// A person with a name, age, and free-form tags.
#[derive(Debug, Clone)]
struct Person {
    name: String,
    age: u32,
    tags: Vec<String>,
}

enum Classification {
    Negative,
    Zero,
    PositiveEven,
    PositiveOdd,
}

impl fmt::Display for Classification {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let s = match self {
            Classification::Negative => "negative",
            Classification::Zero => "zero",
            Classification::PositiveEven => "positive even",
            Classification::PositiveOdd => "positive odd",
        };
        write!(f, "{s}")
    }
}

fn greet(person: &Person, prefix: &str) -> String {
    format!("{prefix}, {}!", person.name)
}

fn classify(n: i64) -> Classification {
    match n {
        n if n < 0 => Classification::Negative,
        0 => Classification::Zero,
        n if n % 2 == 0 => Classification::PositiveEven,
        _ => Classification::PositiveOdd,
    }
}

fn squares(numbers: &[i64]) -> Vec<i64> {
    numbers.iter().map(|x| x * x).filter(|x| *x > 5).collect()
}

fn main() {
    let alice = Person {
        name: "Alice".to_string(),
        age: 30,
        tags: vec!["admin".into(), "early_bird".into()],
    };

    println!("{}", greet(&alice, "Hello"));

    let mut counts: HashMap<String, usize> = HashMap::new();
    for n in -2..=5 {
        let label = classify(n).to_string();
        *counts.entry(label).or_insert(0) += 1;
        println!("{n} -> {}", classify(n));
    }

    println!("squares: {:?}", squares(&[1, 2, 3, 4, 5]));
    println!("counts: {counts:?}");
}
